src/lib/rest/StringFilter.h

```diff
@@ -98,6 +98,20 @@ typedef enum StringFilterValueType
 
 /* ****************************************************************************
 *
+* MatchResult - 
+*/
+typedef enum MatchResult
+{
```

Comment:
Maybe adding a comment with an example for each case (based in a given q/mq filter and a given situation in DB with regards the checked attribute) could clarify each item in the enum for future maintance.

Reply:
Fixed in 65ea8e2 (just a 'see issue 2995')

Reply:
MrDoesntExist and MrIncompatibleTypes maybe are not so evident... Pls, provide me a example of each (q filter + situation at DB) and I will try to provide a suggestion for the comment I mention

Reply:
Well, MrDoesntExist is returned if the attribute/metadata (or compound item) doesn't exist.
E.g., we have an entity E with an attr A and a GET enters for all entities E that have an attribute B>=12.   B doesn't exist in entity E (that has only an attr A), so MrDoesntExist is returned => q/mq with GTE or LTE that before returned !false (== true) now return what they should.

MrNonCompatible is for comparisons of e.g. a string with an integer.

Both these cases returned FALSE before and as GTE uses "!LT", the result was TRUE for GTE.
LTE uses !GT, so, same problem

Reply:
Suggestion:

```
 /* ****************************************************************************
 *
* MatchResult -
*/
typedef enum MatchResult
{
  MrDoesntExist = 0,       // e.g. q=a>2 and 'a' attribute doesn't exist in DB
  MrIncompatibleType,      // e.g. q=a>2 and 'a' attribute exists in DB but is a boolean
  MrNoMatch,               // e.g. q=a>2 and 'a' attribute exists with value (number) 1
  MrMatch                  // e.g. q=a>2 and 'a' attribute exists with value (number) 3
} MatchResult;
```

or something similar.

Reply:
Fixed in 74ae42845